icing/icing_optimisationProofsScript.sml

```diff
@@ -603,11 +603,376 @@ Proof
   \\ fs[]
 QED
 
+fun rename_each [] = ALL_TAC
+| rename_each [first] = qmatch_asmsub_rename_tac first \\ qpat_x_assum first mp_tac \\ disch_then assume_tac
```

Comment:
Is there a particular reason why you are using `qmatch_asmsub_rename_tac` here instead of `rename1`?

Reply:
Yes; `rename1` also looks for matches in the goal. In this case, we only want to rename assumptions.